### PR TITLE
(Graph Beta) Docs for new PSTN and SMS reports

### PR DESCRIPTION
**Pull request by radoslag**
Graph Beta - adding documentation for 3 new APIs:
PSTN blocked users log
Audio Conferencing dial-out usage report
SMS usage log

Updating documentation for 2 existent APIs:
getPstnCalls (new property otherPartyCountryCode)
getDirectRoutingCalls (new properties otherPartyCountryCode, userCountryCode)

CSDL File:
https://dev.azure.com/msazure/One/_git/AD-AggregatorService-Workloads/pullrequest/7527617?_a=files&path=/Workloads/Microsoft.IC3.DataPlatform/override/schema-Prod-beta.csdl

API Review:
https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding/pullrequest/12815

Roadmap item: Feature #16384
https://microsoftgraph.visualstudio.com/Roadmap/_workitems/edit/16384

Validator says that the changelog links are invalid, but they are actually working. The links had been previously pre-generated, but those were broken links, so I manually updated the links to be working.

### CODE REVIEW

**Review by SanderSade — APPROVED** [review]
Approved, with a couple of minor changes suggested offline.

File: api-reference/beta/resources/callrecords-smslogrow.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+# smsLogRow resource type
+
+Namespace: microsoft.graph.callRecords
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents a row of data in the log of a sent or received SMS. Each row maps to one SMS.
+
+Per-country privacy requirements include the obfuscation of the external (not owned by the customer) phone numbers for some countries based on the location of the organization.
+For those countries, the three or four last digits of the phone number are replaced with asterisks (for example, +123 456789***). For an inbound SMS, the sender number may be obfuscated. For an outbound SMS, the receiver number may be obfuscated.
```
Comment (JarbasHorst) [inline api-reference/beta/resources/callrecords-smslogrow.md:19]: For simplicity, I recommend you to merge the last two sentences into one. Suggested revision:

- You can obfuscate the sender/receiver number of an inbound or outbound SMS.

File: api-reference/beta/resources/callrecords-smslogrow.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+|otherPartyCountryCode|String|For an outbound SMS, the country code of the receiver; otherwise (inbound SMS) the country code of the sender. For details, see [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2).|
+|sentDateTime|DateTimeOffset|The date and time when the SMS was sent. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is `2014-01-01T00:00:00Z`.|
+|smsId|String|SMS identifier. Not guaranteed to be unique.|
+|smsType|String|Type of SMS such as outbound or inbound.|
+|smsUnits|Int32|Number of SMS units sent/received.|
+|sourceNumber|String|Partially obfuscated phone number that sent the SMS. For details, see [E.164](https://en.wikipedia.org/wiki/E.164).|
+|tenantCountryCode|String|Country code of the tenant. For details, see [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2).|
+|userCountryCode|String|Country code of the user. For details, see [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2).|
+|userDisplayName|String|Display name of the user.|
+|userId|String|User ID in Graph (GUID).|
```
Comment (JarbasHorst) [inline api-reference/beta/resources/callrecords-smslogrow.md:41]: The user ID usually comes from Azure Active Directory. Microsoft Graph doesn't maintain any database with users. For clarity, I recommend you to rephrase this description. Suggested revision:

- The unique identifier (GUID) for the user in Azure Active Directory.

Same for:
- api-reference/beta/resources/callrecords-directroutinglogrow.md
- api-reference/beta/resources/callrecords-pstnblockeduserslogrow.md
- api-reference/beta/resources/callrecords-pstncalllogrow.md
- api-reference/beta/resources/callrecords-pstnonlinemeetingdialoutreport.md

File: api-reference/beta/resources/callrecords-smslogrow.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+Represents a row of data in the log of a sent or received SMS. Each row maps to one SMS.
+
+Per-country privacy requirements include the obfuscation of the external (not owned by the customer) phone numbers for some countries based on the location of the organization.
+For those countries, the three or four last digits of the phone number are replaced with asterisks (for example, +123 456789***). For an inbound SMS, the sender number may be obfuscated. For an outbound SMS, the receiver number may be obfuscated.
+
+## Properties
+
+|Property|Type|Description|
+|:---|:---|:---|
+|callCharge|Double|Amount of money or cost of the SMS that is charged.|
```
Comment (JarbasHorst) [inline api-reference/beta/resources/callrecords-smslogrow.md:25]: According to the api.md file, this property has the following type: "Edm.Decimal". Is this correct? If yes, then please replace Double with Decimal in lines 25 and 59.

Same for the totalCallCharge here:
- api-reference/beta/resources/callrecords-pstnonlinemeetingdialoutreport.md

File: api-reference/beta/resources/callrecords-pstnblockeduserslogrow.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents a row of data in the public switched telephone network (PSTN) blocked users log. Each row maps to a user blocked/unblocked from making PSTN calls.
+
+## Properties
+
+|Property|Type|Description|
+|:---|:---|:---|
+|blockDateTime|DateTimeOffset|The block or unblock time. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is `2014-01-01T00:00:00Z`.|
```
Comment (JarbasHorst) [inline api-reference/beta/resources/callrecords-pstnblockeduserslogrow.md:22]: What exactly was blocked or unblocked. What is the subject in this sentence? Please, update the description for this property to contain more details about the subject that gets blocked or unblocked.

File: api-reference/beta/resources/callrecords-smslogrow.md (hunk 1 above)
Comment (radoslag) [reply api-reference/beta/resources/callrecords-smslogrow.md]: The sentence was unnecessary, removed.

File: api-reference/beta/resources/callrecords-smslogrow.md (hunk 2 above)
Comment (radoslag) [reply api-reference/beta/resources/callrecords-smslogrow.md]: Agreed with your suggestion, thank you!

File: api-reference/beta/resources/callrecords-smslogrow.md (hunk 3 above)
Comment (radoslag) [reply api-reference/beta/resources/callrecords-smslogrow.md]: Yes, renamed Double to Decimal, on 4 different places.

File: api-reference/beta/resources/callrecords-pstnblockeduserslogrow.md (hunk 4 above)
Comment (radoslag) [reply api-reference/beta/resources/callrecords-pstnblockeduserslogrow.md]: Rephrased the sentence. Thank you for noticing!

File: api-reference/beta/api/callrecords-callrecord-getpstnblockeduserslog.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+|Application| CallRecord-PstnCalls.Read.All, CallRecords.Read.All |
+
+## HTTP request
+
+<!-- {
+  "blockType": "ignored"
+}
+-->
+``` http
+GET /communications/callRecords/getPstnBlockedUsersLog
```
Comment (JarbasHorst) [inline api-reference/beta/api/callrecords-callrecord-getpstnblockeduserslog.md:35]: Can one request this API without both function parameters? If no, I would recommend you include the function parameters to the HTTP request URL. Suggested revision:

- GET /communications/callRecords/getPstnBlockedUsersLog(fromDateTime={fromDateTime},toDateTime={toDateTime})

Same here:
- api-reference/beta/api/callrecords-callrecord-getpstnonlinemeetingdialoutreport.md
- api-reference/beta/api/callrecords-callrecord-getsmslog.md

File: api-reference/beta/api/callrecords-callrecord-getpstnblockeduserslog.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+## Function parameters
+
+In the request URL, provide the following query parameters with values.
+The following table shows the parameters that can be used with this function.
+
+|Parameter|Type|Description|
+|:---|:---|:---|
+|fromDateTime|DateTimeOffset|Start of time range to query. UTC, inclusive.|
```
Comment (JarbasHorst) [inline api-reference/beta/api/callrecords-callrecord-getpstnblockeduserslog.md:45]: Please, add either "Required." or "Optional." to the description sentence to indicate that these parameters are either optional or required.

Same also here:

- api-reference/beta/api/callrecords-callrecord-getpstnonlinemeetingdialoutreport.md
- api-reference/beta/api/callrecords-callrecord-getsmslog.md

**Comment by JarbasHorst** [comment]
Hi @radoslag, thanks for addressing my previous comments. I left additional ones. Please, take a look when you will have time.

File: api-reference/beta/api/callrecords-callrecord-getpstnblockeduserslog.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
```
[the remark this answers is not in the document]
Comment (radoslag) [reply api-reference/beta/api/callrecords-callrecord-getpstnblockeduserslog.md]: UTC replaced with the suggested sentence.

File: api-reference/beta/api/callrecords-callrecord-getpstnblockeduserslog.md (hunk 5 above)
Comment (radoslag) [reply api-reference/beta/api/callrecords-callrecord-getpstnblockeduserslog.md]: The required function parameters added to the HTTP request URL, as suggested.

File: api-reference/beta/api/callrecords-callrecord-getpstnblockeduserslog.md (hunk 6 above)
Comment (radoslag) [reply api-reference/beta/api/callrecords-callrecord-getpstnblockeduserslog.md]: Added "(required parameter)", as suggested.

**Comment by radoslag** [comment]
> Hi @radoslag, thanks for addressing my previous comments. I left additional ones. Please, take a look when you will have time.

Hello @JarbasHorst, thank you very much for your suggestions, I've adjusted the documentation accordingly. Please let me know if otherwise all looks good to you.

**Comment by JarbasHorst** [comment]
Hello @radoslag, I am done with the content review work for your PR but this PR has some API Doctor issues. I asked the team now for support with those issues.

**Comment by Jarbas-MSFT** [comment]
@Lauragra this PR is ready for final review. We had to suppress the API Doctor issues about properties of type decimal because the validation isn't working properly for those types: https://github.com/OneDrive/apidoctor/issues/186

File: changelog/Microsoft.IC3.DataPlatform.json [hunk 8]
```diff
@@ -1,344 +1,434 @@
...
-                    "Target": "clientUserAgent"
-                }
-            ],
-            "Id": "1ab47202-a340-49ba-b47b-f53b7eeb1604",
-            "Cloud": "Prod",
-            "Version": "beta",
-            "CreatedDateTime": "2022-08-24T06:16:18.4859577Z",
-            "WorkloadArea": "Teamwork and communications",
-            "SubArea": "Calls and online meetings"
+  "changelog": [
```
Comment (Lauragra) [inline changelog/Microsoft.IC3.DataPlatform.json:2]: There look to be multiple updates to the changelog file here. To make sure that I validate the entry, can you provide the ID that's associated with this PR?

**Review by Lauragra — APPROVED** [review]

**Pull request merged**